apps/system/sekey_test/sekey_test.c

```diff
@@ -0,0 +1,239 @@
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <tinyara/security_hal.h>
+#include <tinyara/seclink.h>
+#include <mbedtls/x509_crt.h>
+
+#ifdef CONFIG_ECDSA_KEY_SLOT
+#define ECDSA_KEY_SLOT CONFIG_ECDSA_KEY_SLOT
+#else
+#define ECDSA_KEY_SLOT "0,"
+#endif
+
+#ifdef CONFIG_CERT_SLOT
+#define CERT_SLOT CONFIG_CERT_SLOT
+#else
+#define CERT_SLOT "1,"
+#endif
+
+static uint8_t key_test[128] = {
+	0,
+};
+static uint8_t cert_test[128] = {
+	0,
+};
+
+int sekey_test_main(int argc, char *argv[])
+{
+	int i;
+	int ret;
+	int res = 0;
+	char *ptr;
+	sl_ctx shnd;
+	hal_result_e hres = 0;
+	hal_ecdsa_mode ecdsa_mode;
+	hal_data t_hash = {
+		0,
+	};
+	int hash_len = 32;
+	hal_data sign = {
+		0,
+	};
+	int sign_len = 1024;
+
+	mbedtls_x509_crt cert = {
+		0,
+	};
+	hal_data cert_out = {
+		0,
+	};
+	int cert_out_size = 1024;
+	char buf[256] = {
+		0,
+	};
+
+	char key_temp[] = ECDSA_KEY_SLOT;
+	char cert_temp[] = CERT_SLOT;
+	int key_cnt = 0;
+	int cert_cnt = 0;
+
+	ret = sl_init(&shnd);
+	if (ret != SECLINK_OK) {
+		printf("[SECLINK] sl_init fail \n");
+		return -1;
+	}
+
+	ptr = strtok(key_temp, ",");
+	while (ptr != NULL) {
+		key_test[key_cnt++] = atoi(ptr);
+		ptr = strtok(NULL, ",");
+	}
+
+	ptr = strtok(cert_temp, ",");
+	while (ptr != NULL) {
+		cert_test[cert_cnt++] = atoi(ptr);
+		ptr = strtok(NULL, ",");
+	}
+
+	t_hash.data = malloc(hash_len);
+	if (t_hash.data == NULL) {
+		res = -1;
+		goto exit;
+	}
+	memset(t_hash.data, 0xa5, hash_len);
+	t_hash.data_len = hash_len;
+
+	sign.data = malloc(sign_len);
+	if (sign.data == NULL) {
+		res = -1;
+		goto exit;
+	}
+	memset(sign.data, 0, sign_len);
+
+	ecdsa_mode.curve = HAL_ECDSA_SEC_P256R1;
+	ecdsa_mode.hash_t = HAL_HMAC_SHA256;
+
+	printf("Start Key & Cert Checking ...\n");
+
+	for (i = 0; i < key_cnt; i++) {
+
+		printf("[Key] ECDSA Sign Slot#%d ... ", key_test[i]);
+		ret = sl_ecdsa_sign_md(shnd, ecdsa_mode, &t_hash, key_test[i], &sign, &hres);
+		if (ret != SECLINK_OK || hres != HAL_SUCCESS) {
+			printf("[ FAIL ] / Slot #%d / [%d][%d] \n", key_test[i], ret, hres);
+			res = -1;
+			goto exit;
+		}
+		printf("[ OK ]\n");
+
+		printf("[Key] ECDSA Verify Slot#%d ... ", key_test[i]);
+		ret = sl_ecdsa_verify_md(shnd, ecdsa_mode, &t_hash, &sign, key_test[i], &hres);
+		if (ret != SECLINK_OK || hres != HAL_SUCCESS) {
+			printf("[ FAIL ] / Slot #%d / [%d][%d] \n", key_test[i], ret, hres);
+			res = -1;
+			goto exit;
+		}
+		printf("[ OK ] \n");
+	}
+
+	cert_out.data = malloc(cert_out_size);
+	if (cert_out.data == NULL) {
+		res = -1;
+		goto exit;
+	}
+
+	for (i = 0; i < cert_cnt; i++) {
+		memset(buf, 0, sizeof(buf));
+		memset(cert_out.data, 0, cert_out_size);
+		cert_out.data_len = cert_out_size;
+
+		printf("[Cert] Get Cert Slot#%d ... ", cert_test[i]);
+		ret = sl_get_certificate(shnd, cert_test[i], &cert_out, &hres);
+		if (ret != SECLINK_OK || hres != HAL_SUCCESS) {
+			printf("[ FAIL ] / ret : %d, hres : %d \n", ret, hres);
```

Comment:
We cannot distinguish the failure between line 190 and 201.
How about remaining the reason in detail?

Reply:
In this case, there are several types of errors so we cannot explain the reason in one sentence.
The only thing that users can know is which command is not working in the board.
For more details, the user should check 'hres' value at seclink.